chat-components/src/services/BroadcastService.ts

```diff
@@ -14,8 +14,66 @@ const broadcastServiceSubList: Record<string, any> = {};
 let pubChannel: any;
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 let subChannel: any;
+let eventQueue: EventQueue;
+
+class EventQueue {
+    private queueing: boolean = true;
+    private channelEventQueue: ICustomEvent[];
+    private queuingId?: NodeJS.Timeout;
+
+    constructor() {
+        this.channelEventQueue = [];
+        this.queueEvents();
+    }
+
+    processEvents() {
+        while (this.channelEventQueue.length > 0) {
+            const event = this.channelEventQueue.shift();
+            if (event) {
+                pubChannel.postMessage(event);
+            }
+        }
+    }
+
+    queueEvents(timeout = 3000) {
+        this.processEvents();
+        if (this.queueing) {
+            this.queuingId = setTimeout(() => {
+                this.queueEvents();
+            }, timeout);
+        }
+    }
+
+    resumeQueueing() {
+        this.queueing = true;
+    }
+
+    stopQueueing() {
+        this.queueing = false;
+        this.queuingId = undefined
+    }
+
+    pushEvent(event: any) {
+        this.channelEventQueue.push(event);
+    }
+
+    popEvent(event: any) {
```

Comment:
lets use a Map<Integer, Object> so we access to the id in O(1) instead of filtering O(n)